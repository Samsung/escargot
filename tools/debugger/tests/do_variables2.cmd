b do_variables2.js:32
c
variables
c
