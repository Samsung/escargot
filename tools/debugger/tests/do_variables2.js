/*
 * Copyright (c) 2020-present Samsung Electronics Co., Ltd
 *
 *  This library is free software; you can redistribute it and/or
 *  modify it under the terms of the GNU Lesser General Public
 *  License as published by the Free Software Foundation; either
 *  version 2.1 of the License, or (at your option) any later version.
 *
 *  This library is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 *  Lesser General Public License for more details.
 *
 *  You should have received a copy of the GNU Lesser General Public
 *  License along with this library; if not, write to the Free Software
 *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301
 *  USA
 */

{
  let a;
  let b = null;
  let c = true;
  let d = false;
  let e = 3.14;
  let f = "str";
  let g = Symbol("sym");
  let h = {};
  let i = [];
  let j = function() {};
  let l = 900719925474099123n; // Number.MAX_SAFE_INTEGER x 100 + 23

  j();

  let k; // unaccessible (because it has not been initialized yet)
}
