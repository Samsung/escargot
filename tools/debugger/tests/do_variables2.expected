Connecting to: localhost:6501
Connection created!!!
Stopped at tools/debugger/tests/do_variables2.js:22
(escargot-debugger) b do_variables2.js:32
Breakpoint 1 at tools/debugger/tests/do_variables2.js:32
(escargot-debugger) c
Stopped at breakpoint:1 tools/debugger/tests/do_variables2.js:32
(escargot-debugger) variables
a: undefined
b: null
c: boolean: true
d: boolean: false
e: number: 3.14
f: string: str
g: symbol: sym
h: object
i: array
j: function
k: unaccessible
(escargot-debugger) c
Connection closed.
